### ISSUE TITLE
Learning to Diagnose with LSTM Recurrent Neural Networks

### ISSUE THREAD

**Issue by ghuijo** [yskim0/One-Day-One-Paper#77]
본 논문은 2015년 등재 이후 revised된 2017년 버전임.
ICLR에는 2016년에 published됨.

## Abstract
* ICU 환자 데이터와 같은 EHR(Electronic Health Record) data (sensor data, lab test results 등) 는 irregular sampling, missing data로 다루기 어려움.
* 본 논문에서 LSTM이 clinical measurements의 multivariate time series에서 patterns을 인식하는 능력을 평가할 것임.
* 또한, straightforward하고 효율적인 training strategy를 보여줄 것임.

## Introduction
* PICU (pediatric intensive care unit)에서 측정된 time series data의 potential uses
  * classifying diagnoses accurately / predicting length of stay / predicting future illness / predicting mortality
* 위 dataset의 문제점
  * vary in length with stays (진짜 공감) / 관찰 data(sensor data, vital signs, lab test results, subjective assessments) 가 irregularly하게 sampled되어 있고, missing values 때문에 plagued됨. (완전 공감)
  * long-term time dependencies complicate learning with many algorithms
* LSTM은 long range dependencies와 nonlinear dynamics를 capture할 수 있음.
* 다른 모델들과 달리 nn은 representations를 학습하고, unforeseen structure를 discover할 수 있음.
* 429개의 진단 코드 중에서 most common한 128개의 코드에 집중함. (따로 feature selection 과정은 안 거친 듯)
* strong baseline은 fixed window와 hand-engineered features로 학습시킴. (비교 용도)
* RNN은 straightforward target replication strategy로 학습시킴. + 추가로 환자의 additional information을 써서 효과적인 지 평가도 하고, non-recurrent connections에 dropout 적용도 해봄.
* MLP trained on hand-engineered features(위에서 언급한 strong baseline)의 성능을 LSTMs with target replication and dropout이 능가했음.

## Related Work
* LSTM RNNs / NN for medical data / NN for multilabel classification / machine learning for clinical time series / target replication / regularizing RNNs / key differences

## Data Description
* collection of anonymized clinical time series extracted from the EHR system at Children's Hospital LA
* 10, 401 PICU episodes, each a multivariate time series of 13 variables
  * diastolic and systolic blood pressure, peripheral capillary refill rate, end-tidal CO2, fraction of inspired O2, Glascow coma scale, blood glucose, heart rate, pH, respiratory rate, blood oxygen saturation, body temperature, urine output.
  * Episodes vary in length from 12 hours to several months
* resample all time series to **_hourly rate_**, taking the **_mean_** measurement within each one hour window
* forward-filling & backward-filling 써서 window-based resampling 과정에서 생겨난 gap을 채움.
* clinically normal value를 impute했는데 이 value는 domain experts가 정의내린 것(전문의를 뜻하는 듯)
* 그런데 back-filling이 forecasting에 적절하지는 않음.
* 모든 variables를 0과 1 사이 값으로 rescale함.
* 진단 코드는429개 중 128개를 쓰는데, data에서 50번 이상 나온 것들임.
* 진단 코드가 clinicians에 의해 assign된 것이므로 LSTM-based 진단 시스템 vs human experts 비교가 됨.

## Methods
* 목적이 recognizing diagnoses인데, observable physiologic characteristic of patients와 phenotyping도 함.
* T는 sequence 길이이고, x^(1), ..., x^(T)는 series of observation임.
* 본 논문에서 제안한 LSTM RNN은 forget gates가 있는 memory cells은 쓰나 peephole connections은 쓰지 않음.
* highest LSTM layer 위에 fc layer를 쓰고, element-wise sigmoid activation function 사용.
* loss function으로 log loss를 쓰는데 매 output마다 적용함.
<image src="https://user-images.githubusercontent.com/67627471/129898161-c49d7dc5-5a85-44da-af6e-60e090f1100e.png" width="40%" height="40%" />

#### Sequential Target Replication
* 기존 simple RNN 모델
![image](https://user-images.githubusercontent.com/67627471/129898522-22002260-b926-436a-9120-48af9c046b29.png)

* 저렇게 최종 output만 받으면 information across many sequence stepts를 놓치게 됨. 그래서 local error signal을 매 스텝마다 줌.
![image](https://user-images.githubusercontent.com/67627471/129898714-9e74f401-2a22-4222-af98-c2acbae93eb9.png)

* training 동안 model이 errors를 intermediate targets에서 각 sequence step으로 backpropagate함.
![image](https://user-images.githubusercontent.com/67627471/129898771-9b0558e2-5631-44b3-881e-880cbf80cd67.png)

#### Auxiliary Output Training
* 기존에 제외했던 진단 코드 301 labels를 overfitting을 막기 위한 수단으로 활용함.

#### Regularization
* overfitting 방지로 target replication과 auxiliary outputs 사용했는데 추가로 l2 weight decay와 dropout도 씀.
* compute each hidden layer’s sequence of activations in the left-to-right direction -> apply dropout before computing the next layer’s activations

#### Experiments
* 80% / 10% / 10% : training / test / validate
* 100 epochs using SGD with momentum

## Results
<image src="https://user-images.githubusercontent.com/67627471/129899774-a9434461-e37b-4fc7-be9e-6d05b6aa151c.png" width="70%" height="70%" />

<image src="https://user-images.githubusercontent.com/67627471/129899848-c5da49af-0d8b-4d44-8128-2f522e629e21.png" width="70%" height="70%" />

##
* hand-engineered features?
* peephole connections?
